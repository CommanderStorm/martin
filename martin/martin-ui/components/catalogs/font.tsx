import { Download, Eye, Search, Type } from "lucide-react";
import { ErrorState } from "@/components/error/error-state";
import { CatalogSkeleton } from "@/components/loading/catalog-skeleton";
import { Badge } from "@/components/ui/badge";
import { Button } from "@/components/ui/button";
import { Card, CardContent, CardDescription, CardHeader, CardTitle } from "@/components/ui/card";
import { Input } from "@/components/ui/input";
import type { Font } from "@/lib/types";
import { DisabledNonInteractiveButton } from "../ui/disabledNonInteractiveButton";
import { Tooltip, TooltipContent, TooltipTrigger } from "../ui/tooltip";

interface FontCatalogProps {
  fonts?: { [name: string]: Font };
  searchQuery?: string;
  onSearchChangeAction?: (query: string) => void;
  isLoading?: boolean;
  error?: Error | null;
  onRetry?: () => void;
  isRetrying?: boolean;
}

export function FontCatalog({
  fonts,
  searchQuery = "",
  onSearchChangeAction = () => {},
  isLoading,
  error = null,
  onRetry,
  isRetrying = false,
}: FontCatalogProps) {
  if (isLoading) {
    return <CatalogSkeleton description="Preview all available font assets" title="Font Catalog" />;
  }

  if (error) {
    return (
      <ErrorState
        description="Unable to fetch font catalog from the server"
        error={error}
        isRetrying={isRetrying}
        onRetry={onRetry}
        showDetails={true}
        title="Failed to Load Fonts"
        variant="server"
      />
    );
  }

  const filteredFonts = Object.entries(fonts || {}).filter(
    ([name, font]) =>
      name.toLowerCase().includes(searchQuery.toLowerCase()) ||
      font.family.toLowerCase().includes(searchQuery.toLowerCase()) ||
      font.style.toLowerCase().includes(searchQuery.toLowerCase()),
  );

  return (
    <div className="space-y-6">
      <div className="flex items-center justify-between">
        <div>
          <h2 className="text-2xl font-bold text-foreground">Font Catalog</h2>
          <p className="text-muted-foreground">Preview all available font assets</p>
        </div>
        <div className="relative">
          <Search className="absolute left-3 top-1/2 transform -translate-y-1/2 text-gray-400 w-4 h-4" />
          <Input
            className="pl-10 w-64 bg-card"
            onChange={(e) => onSearchChangeAction(e.target.value)}
            placeholder="Search fonts..."
            value={searchQuery}
          />
        </div>
      </div>

      <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6">
        {filteredFonts.map(([name, font]) => (
          <Card className="hover:shadow-lg transition-shadow" key={name}>
            <CardHeader>
              <div className="flex items-center justify-between">
                <div className="flex items-center space-x-2">
                  <Type className="w-5 h-5 text-primary" />
                  <CardTitle className="text-lg">{name}</CardTitle>
                </div>
                {font.format && (
                  <Badge className="uppercase" variant="secondary">
                    {font.format}
                  </Badge>
                )}
              </div>
              <CardDescription>
                Family: {font.family} â€¢ Style: {font.style}
              </CardDescription>
            </CardHeader>
            <CardContent>
              <div className="space-y-4">
                <div className="p-3 bg-gray-50 text-gray-900 rounded-lg">
                  <p className="text-sm font-medium mb-2 text-gray-900">Preview:</p>
                  <Tooltip>
                    <TooltipTrigger>
                      <p
                        className="text-base text-gray-900 blur-sm animate-pulse"
                        style={{ fontFamily: font.family, fontWeight: 500 }}
                      >
                        The quick brown fox jumps over the lazy dog
                      </p>
                    </TooltipTrigger>
                    <TooltipContent>
                      <p>Not currently implemented in the frontend</p>
                    </TooltipContent>
                  </Tooltip>
                </div>
                <div className="space-y-2 text-sm text-muted-foreground">
                  <div className="flex justify-between">
                    <span>Glyph count:</span>
                    <span>{font.glyphs}</span>
                  </div>
                </div>
                <div className="flex space-x-2">
                  <Button className="flex-1 bg-transparent" size="sm" variant="outline">
                    <Download className="w-4 h-4 mr-2" />
                    Download
                  </Button>
                  <Tooltip>
                    <TooltipTrigger className="flex-1 flex">
                      <DisabledNonInteractiveButton className="flex-1 grow" disabled size="sm">
                        <Eye className="w-4 h-4 mr-2" />
                        Details
                      </DisabledNonInteractiveButton>
                    </TooltipTrigger>
                    <TooltipContent>
                      <p>Not currently implemented in the frontend</p>
                    </TooltipContent>
                  </Tooltip>
                </div>
              </div>
            </CardContent>
          </Card>
        ))}
      </div>

      {filteredFonts.length === 0 && searchQuery && (
        <div className="text-center py-12">
          <p className="text-muted-foreground">No fonts found matching "{searchQuery}"</p>
        </div>
      )}
    </div>
  );
}
