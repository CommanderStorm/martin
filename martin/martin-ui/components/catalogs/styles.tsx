import { Brush, Download, Eye, Map, Search } from "lucide-react";
import { ErrorState } from "@/components/error/error-state";
import { CatalogSkeleton } from "@/components/loading/catalog-skeleton";
import { Badge } from "@/components/ui/badge";
import { Button } from "@/components/ui/button";
import {
	Card,
	CardContent,
	CardDescription,
	CardHeader,
	CardTitle,
} from "@/components/ui/card";
import { Input } from "@/components/ui/input";
import { DisabledNonInteractiveButton } from "../ui/disabledNonInteractiveButton";
import { Tooltip, TooltipContent, TooltipTrigger } from "../ui/tooltip";
import type { Style } from "@/lib/types";

interface StylesCatalogProps {
	styles?: {[name: string]: Style};
	searchQuery?: string;
	onSearchChangeAction?: (query: string) => void;
	isLoading?: boolean;
	error?: Error | null;
	onRetry?: () => void;
	isRetrying?: boolean;
}

export function StylesCatalog({
	styles,
	searchQuery = "",
	onSearchChangeAction = () => {},
	isLoading,
	error = null,
	onRetry,
	isRetrying = false,
}: StylesCatalogProps) {
	if (isLoading) {
		return (
			<CatalogSkeleton
				title="Styles Catalog"
				description="Browse and preview all available map styles and themes"
			/>
		);
	}

	if (error) {
		return (
			<ErrorState
				title="Failed to Load Styles"
				description="Unable to fetch style catalog from the server"
				error={error}
				onRetry={onRetry}
				isRetrying={isRetrying}
				variant="server"
				showDetails={true}
			/>
		);
	}

	const filteredStyles = Object.entries(styles ||{}).filter(
		([name,style]) =>
			name.toLowerCase().includes(searchQuery.toLowerCase()) ||
			style.description.toLowerCase().includes(searchQuery.toLowerCase()) ||
			style.type.toLowerCase().includes(searchQuery.toLowerCase()),
	);

	return (
		<div className="space-y-6">
			<div className="flex items-center justify-between">
				<div>
					<h2 className="text-2xl font-bold text-foreground">Styles Catalog</h2>
					<p className="text-muted-foreground">
						Browse and preview all available map styles and themes
					</p>
				</div>
				<div className="relative">
					<Search className="absolute left-3 top-1/2 transform -translate-y-1/2 text-gray-400 w-4 h-4" />
					<Input
						placeholder="Search styles..."
						className="pl-10 w-64 bg-card"
						value={searchQuery}
						onChange={(e) => onSearchChangeAction(e.target.value)}
					/>
				</div>
			</div>

			<div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6">
				{filteredStyles.map(([name,style]) => (
					<Card key={name} className="hover:shadow-lg transition-shadow">
						<CardHeader>
							<div className="flex items-center justify-between">
								<div className="flex items-center space-x-2">
									<Brush className="w-5 h-5 text-primary" />
									<CardTitle className="text-lg">{name}</CardTitle>
								</div>
								<Badge variant="secondary">{style.type}</Badge>
							</div>
							<CardDescription>{style.description}</CardDescription>
						</CardHeader>
						<CardContent>
							<div className="space-y-4">
								<div className="p-3 aspect-video rounded-lg bg-gradient-to-br from-gray-200 to-gray-300 flex items-center justify-center relative overflow-hidden">
									<div className="absolute inset-0 opacity-20">
										<div
											className="w-full h-full bg-gradient-to-r"
											style={{
												background: `linear-gradient(45deg, ${style.colors.join(", ")})`,
											}}
										></div>
									</div>
									<Map className="w-8 h-8 text-gray-600 z-10" />
								</div>
								<div className="space-y-2 text-sm text-muted-foreground">
									<div className="flex justify-between">
										<span>Version:</span>
										<span>{style.version}</span>
									</div>
									<div className="flex justify-between">
										<span>Layers:</span>
										<span>{style.layers}</span>
									</div>
									<div className="flex justify-between">
										<span>Usage:</span>
										<span>{style.usage}</span>
									</div>
									<div className="flex justify-between">
										<span>Modified:</span>
										<span>{style.lastModified}</span>
									</div>
								</div>
								<div>
									<p className="text-sm font-medium mb-2">Color Palette:</p>
									<div className="flex space-x-1">
										{style.colors.map((color, i) => (
											<div
												key={i}
												className="w-6 h-6 rounded border border-gray-200"
												style={{ backgroundColor: color }}
												title={color}
											></div>
										))}
									</div>
								</div>
								<div className="flex space-x-2">
									<Button
										size="sm"
										variant="outline"
										className="flex-1 bg-transparent"
									>
										<Download className="w-4 h-4 mr-2" />
										Download
									</Button>

									<Tooltip>
										<TooltipTrigger className="flex flex-1">
											<DisabledNonInteractiveButton
												size="sm"
												className="flex-1"
											>
												<Eye className="w-4 h-4 mr-2" />
												Preview
											</DisabledNonInteractiveButton>
										</TooltipTrigger>
										<TooltipContent>
											<p>Not currently implemented in the frontend</p>
										</TooltipContent>
									</Tooltip>
								</div>
							</div>
						</CardContent>
					</Card>
				))}
			</div>

			{filteredStyles.length === 0 && searchQuery && (
				<div className="text-center py-12">
					<p className="text-muted-foreground">
						No styles found matching "{searchQuery}"
					</p>
				</div>
			)}
		</div>
	);
}
