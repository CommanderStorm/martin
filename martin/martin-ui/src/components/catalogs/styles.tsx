import { Brush, Eye, Map, Search } from "lucide-react";

import { ErrorState } from "@/components/error/error-state";
import { CatalogSkeleton } from "@/components/loading/catalog-skeleton";
import { Badge } from "@/components/ui/badge";
import { Button } from "@/components/ui/button";
import { Card, CardContent, CardDescription, CardHeader, CardTitle } from "@/components/ui/card";
import { Input } from "@/components/ui/input";
import type { Style } from "@/lib/types";
import { DisabledNonInteractiveButton } from "../ui/disabledNonInteractiveButton";
import { Tooltip, TooltipContent, TooltipTrigger } from "../ui/tooltip";
import { CopyLinkButton } from "../ui/copy-link-button";
import { buildMartinUrl } from "@/lib/api";
import 'maplibre-gl/dist/maplibre-gl.css';
import {Map as MapLibreMap} from '@vis.gl/react-maplibre';
import {useState} from "react"

interface StylesCatalogProps {
  styles?: { [name: string]: Style };
  searchQuery?: string;
  onSearchChangeAction?: (query: string) => void;
  isLoading?: boolean;
  error?: Error | null;
  onRetry?: () => void;
  isRetrying?: boolean;
}

export function StylesCatalog({
  styles,
  searchQuery = "",
  onSearchChangeAction = () => {},
  isLoading,
  error = null,
  onRetry,
  isRetrying = false,
}: StylesCatalogProps) {
  const [viewState, setViewState] = useState({
    longitude: -100,
    latitude: 40,
    zoom: 3.5
  });
  if (isLoading) {
    return (
      <CatalogSkeleton
        description="Preview all available map styles and themes"
        title="Styles Catalog"
      />
    );
  }

  if (error) {
    return (
      <ErrorState
        description="Unable to fetch style catalog from the server"
        error={error}
        isRetrying={isRetrying}
        onRetry={onRetry}
        showDetails={true}
        title="Failed to Load Styles"
        variant="server"
      />
    );
  }

  const filteredStyles = Object.entries(styles || {}).filter(
    ([name, style]) =>
      name.toLowerCase().includes(searchQuery.toLowerCase()) ||
      style.path.toLowerCase().includes(searchQuery.toLowerCase()) ||
      style.type?.toLowerCase().includes(searchQuery.toLowerCase()),
  );

  return (
    <div className="space-y-6">
      <div className="flex items-center justify-between">
        <div>
          <h2 className="text-2xl font-bold text-foreground">Styles Catalog</h2>
          <p className="text-muted-foreground">
            Browse and preview all available map styles and themes
          </p>
        </div>
        <div className="relative">
          <Search className="absolute left-3 top-1/2 transform -translate-y-1/2 text-gray-400 w-4 h-4" />
          <Input
            className="pl-10 w-64 bg-card"
            onChange={(e) => onSearchChangeAction(e.target.value)}
            placeholder="Search styles..."
            value={searchQuery}
          />
        </div>
      </div>

      <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6">
        {filteredStyles.map(([name, style]) => (
          <Card className="hover:shadow-lg transition-shadow" key={name}>
            <CardHeader>
              <div className="flex items-center justify-between">
                <div className="flex items-center space-x-2">
                  <Brush className="w-5 h-5 text-primary" />
                  <CardTitle className="text-lg">{name}</CardTitle>
                </div>
                {style.type && <Badge variant="secondary">{style.type}</Badge>}
              </div>
              <CardDescription>{style.path}</CardDescription>
            </CardHeader>
            <CardContent>
              <div className="space-y-4">
                <MapLibreMap
                    reuseMaps
                    {...viewState}
                    onMove={evt => setViewState(evt.viewState)}
                    style={{
                      width: "100%",
                      aspectRatio: 16/9,
                      borderRadius: "var(--radius)",
                      backgroundImage: "linear-gradient(to bottom right, var(--tw-gradient-stops))",
                      backgroundColor: "#E5E7EB",
                    }}
                    mapStyle={buildMartinUrl(`/style/${name}`)}
                  />
                <div className="space-y-2 text-sm text-muted-foreground">
                  {style.versionHash && (
                    <div className="flex justify-between">
                      <span>Version:</span>
                      <span>{style.versionHash}</span>
                    </div>
                  )}
                  {style.layerCount && (
                    <div className="flex justify-between">
                      <span>Layers:</span>
                      <span>{style.layerCount}</span>
                    </div>
                  )}
                  {style.lastModifiedAt && (
                    <div className="flex justify-between">
                      <span>Modified:</span>
                      <span>{style.lastModifiedAt?.toLocaleString()}</span>
                    </div>
                  )}
                </div>
                {style.colors && (
                  <div>
                    <p className="text-sm font-medium mb-2">Color Palette:</p>
                    <div className="flex space-x-1">
                      {style.colors.map((color, i) => (
                        <div
                          className="w-6 h-6 rounded border border-gray-200"
                          key={i}
                          style={{ backgroundColor: color }}
                          title={color}
                        ></div>
                      ))}
                    </div>
                  </div>
                )}
                <div className="flex space-x-2">
                  <CopyLinkButton
                    className="flex-1 bg-transparent"
                    size="sm"
                    variant="outline"
                    link={buildMartinUrl(`/style/${name}`)}
                    toastMessage="Style link copied!"
                  />

                  <Tooltip>
                    <TooltipTrigger className="flex flex-1">
                      <DisabledNonInteractiveButton className="flex-1" size="sm">
                        <Eye className="w-4 h-4 mr-2" />
                        Preview
                      </DisabledNonInteractiveButton>
                    </TooltipTrigger>
                    <TooltipContent>
                      <p>Not currently implemented in the frontend</p>
                    </TooltipContent>
                  </Tooltip>
                </div>
              </div>
            </CardContent>
          </Card>
        ))}
      </div>

      {filteredStyles.length === 0 && (
        <div className="text-center py-12">
          <p className="text-muted-foreground mb-2">
            {searchQuery
              ? `No styles found matching "${searchQuery}"`
              : "No styles found."
            }
          </p>
          <Button
            asChild
            variant="link"
            size="lg"
          >
            <a
              href="https://maplibre.org/martin/sources-styles.html"
              target="_blank"
              rel="noopener noreferrer"
            >
              Learn how to configure Styles
            </a>
          </Button>
        </div>
      )}
    </div>
  );
}
