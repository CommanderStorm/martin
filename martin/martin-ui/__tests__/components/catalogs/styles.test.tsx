import { fireEvent, render, screen } from "@testing-library/react";
import type React from "react";
import { StylesCatalog } from "@/components/catalogs/styles";
import type { Style } from "@/lib/types";

// Mock all dependencies
jest.mock("@/components/error/error-state", () => ({
  ErrorState: ({ title, description }: { title: string; description: string }) => (
    <div data-testid="error-state">
      <div data-testid="error-title">{title}</div>
      <div data-testid="error-description">{description}</div>
    </div>
  ),
}));

jest.mock("@/components/loading/catalog-skeleton", () => ({
  CatalogSkeleton: ({ title, description }: { title: string; description: string }) => (
    <div data-testid="catalog-skeleton">
      <div data-testid="skeleton-title">{title}</div>
      <div data-testid="skeleton-description">{description}</div>
    </div>
  ),
}));

// Mock UI components to avoid tooltip provider issues
jest.mock("@/components/ui/tooltip", () => ({
  Tooltip: ({ children }: { children: React.ReactNode }) => <div>{children}</div>,
  TooltipContent: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="tooltip-content">{children}</div>
  ),
  TooltipTrigger: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="tooltip-trigger">{children}</div>
  ),
}));

jest.mock("@/components/ui/button", () => ({
  Button: ({ asChild, children, ...props }: any) => <button {...props}>{children}</button>,
}));

jest.mock("@/components/ui/copy-link-button", () => ({
  CopyLinkButton: ({ toastMessage, children, ...props }: any) => (
    <button data-testid="copy-link-button" {...props}>{children ?? "Copy Link"}</button>
  ),
}));

jest.mock("@/components/ui/badge", () => ({
  Badge: ({ children, ...props }: any) => (
    <span data-testid="badge" {...props}>
      {children}
    </span>
  ),
}));

jest.mock("@/components/ui/input", () => ({
  Input: (props: any) => <input {...props} />,
}));

jest.mock("@/components/ui/card", () => ({
  Card: ({ children, ...props }: any) => <div {...props}>{children}</div>,
  CardContent: ({ children, ...props }: any) => (
    <div data-testid="card-content" {...props}>
      {children}
    </div>
  ),
  CardDescription: ({ children, ...props }: any) => (
    <div data-testid="card-description" {...props}>
      {children}
    </div>
  ),
  CardHeader: ({ children, ...props }: any) => (
    <div data-testid="card-header" {...props}>
      {children}
    </div>
  ),
  CardTitle: ({ children, ...props }: any) => (
    <div data-testid="card-title" {...props}>
      {children}
    </div>
  ),
}));

jest.mock("@/components/ui/disabledNonInteractiveButton", () => ({
  DisabledNonInteractiveButton: ({ children, ...props }: any) => (
    <button {...props} disabled>
      {children}
    </button>
  ),
}));

jest.mock("lucide-react", () => ({
  Brush: () => <div data-testid="brush-icon">Brush</div>,
  Download: () => <div data-testid="download-icon">Download</div>,
  Eye: () => <div data-testid="eye-icon">Eye</div>,
  Map: () => <div data-testid="map-icon">Map</div>,
  Search: () => <div data-testid="search-icon">Search</div>,
}));

// Mock MapLibre to avoid dynamic import issues
jest.mock("@vis.gl/react-maplibre", () => ({
  Map: ({ children, ...props }: any) => (
    <div data-testid="maplibre-map" style={props.style}>
      {children}
    </div>
  ),
}));

// Mock maplibre-gl CSS import
jest.mock("maplibre-gl/dist/maplibre-gl.css", () => ({}));

describe("StylesCatalog Component", () => {
  const mockStyles: { [name: string]: Style } = {
    "Basic Style": {
      colors: ["#FF5733", "#33FF57", "#3357FF", "#F3FF33"],
      lastModifiedAt: new Date("2023-01-15"),
      layerCount: 10,
      path: "/styles/basic.json",
      type: "vector",
    },
    "Hybrid Style": {
      lastModifiedAt: new Date("2023-03-25"),
      layerCount: 15,
      path: "/styles/hybrid.json",
      type: "hybrid",
      versionHash: "abc123",
    },
    "Satellite Style": {
      path: "/styles/satellite.json",
    },
  };

  const defaultProps = {
    error: null,
    isLoading: false,
    isRetrying: false,
    onRetry: jest.fn(),
    onSearchChangeAction: jest.fn(),
    searchQuery: "",
    styles: mockStyles,
  };

  it("matches snapshot for loading state", () => {
    const { asFragment } = render(<StylesCatalog {...defaultProps} isLoading={true} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot for loaded state with mock data", () => {
    const { asFragment } = render(<StylesCatalog {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders loading skeleton when isLoading is true", () => {
    render(<StylesCatalog {...defaultProps} isLoading={true} />);
    expect(screen.getByTestId("catalog-skeleton")).toBeInTheDocument();
    expect(screen.getByTestId("skeleton-title").textContent).toBe("Styles Catalog");
    expect(screen.getByTestId("skeleton-description").textContent).toBe(
      "Preview all available map styles and themes",
    );
  });

  it("renders error state when there is an error", () => {
    const error = new Error("Test error");
    render(<StylesCatalog {...defaultProps} error={error} />);
    expect(screen.getByTestId("error-state")).toBeInTheDocument();
    expect(screen.getByTestId("error-title").textContent).toBe("Failed to Load Styles");
  });

  it("renders styles catalog correctly", () => {
    render(<StylesCatalog {...defaultProps} />);
    expect(screen.getByText("Styles Catalog")).toBeInTheDocument();

    // Get all card headers
    const headers = screen.getAllByTestId("card-header");
    expect(headers.length).toBe(3);

    // Check that each style name is displayed
    expect(screen.getByText("Basic Style")).toBeInTheDocument();
    expect(screen.getByText("Satellite Style")).toBeInTheDocument();
    expect(screen.getByText("Hybrid Style")).toBeInTheDocument();

    // Verify paths are displayed
    expect(screen.getByText("/styles/basic.json")).toBeInTheDocument();
    expect(screen.getByText("/styles/satellite.json")).toBeInTheDocument();
    expect(screen.getByText("/styles/hybrid.json")).toBeInTheDocument();

    // Verify type badges are displayed
    const badges = screen.getAllByTestId("badge");
    expect(badges.length).toBe(2);
    expect(badges[0].textContent).toBe("vector");
    expect(badges[1].textContent).toBe("hybrid");

    // Verify layer counts are displayed
    expect(screen.getByText("10")).toBeInTheDocument();
    expect(screen.getByText("15")).toBeInTheDocument();

    // Verify version hashes are displayed
    expect(screen.getByText("abc123")).toBeInTheDocument();
  });

  it("filters styles based on search query - by name", () => {
    render(<StylesCatalog {...defaultProps} searchQuery="basic" />);

    // Should only show the Basic Style
    const headers = screen.getAllByTestId("card-header");
    expect(headers.length).toBe(1);
    expect(screen.getByText("Basic Style")).toBeInTheDocument();
    expect(screen.queryByText("Satellite Style")).not.toBeInTheDocument();
    expect(screen.queryByText("Hybrid Style")).not.toBeInTheDocument();
  });

  it("filters styles based on search query - by path", () => {
    render(<StylesCatalog {...defaultProps} searchQuery="satellite.json" />);

    // Should only show the Satellite Style
    const headers = screen.getAllByTestId("card-header");
    expect(headers.length).toBe(1);
    expect(screen.queryByText("Basic Style")).not.toBeInTheDocument();
    expect(screen.getByText("Satellite Style")).toBeInTheDocument();
    expect(screen.queryByText("Hybrid Style")).not.toBeInTheDocument();
  });

  it("filters styles based on search query - by type", () => {
    render(<StylesCatalog {...defaultProps} searchQuery="hybrid" />);

    // Should only show the Hybrid Style
    const headers = screen.getAllByTestId("card-header");
    expect(headers.length).toBe(1);
    expect(screen.queryByText("Basic Style")).not.toBeInTheDocument();
    expect(screen.queryByText("Satellite Style")).not.toBeInTheDocument();
    expect(screen.getByText("Hybrid Style")).toBeInTheDocument();
  });

  it("shows no results message when search has no matches", () => {
    render(<StylesCatalog {...defaultProps} searchQuery="nonexistent" />);
    expect(screen.getByText(/No styles found matching "nonexistent"/i)).toBeInTheDocument();

    // Should not render any cards
    const headers = screen.queryAllByTestId("card-header");
    expect(headers.length).toBe(0);
  });

  it("calls onSearchChangeAction when search input changes", () => {
    render(<StylesCatalog {...defaultProps} />);
    const searchInput = screen.getByPlaceholderText("Search styles...");

    fireEvent.change(searchInput, { target: { value: "new search" } });
    expect(defaultProps.onSearchChangeAction).toHaveBeenCalledWith("new search");
  });

  it("displays color palettes when available", () => {
    render(<StylesCatalog {...defaultProps} />);

    // We should have color palettes for each style
    const colorPalettes = screen.getAllByText("Color Palette:");
    expect(colorPalettes.length).toBe(1);

    // Check that not implemented buttons have a tooltip
    const mapPreviews = screen.getAllByTestId("tooltip-trigger");
    expect(mapPreviews.length).toBe(3);

    // Each preview button should contain an eye icon
    const eyeIcons = screen.getAllByTestId("eye-icon");
    expect(eyeIcons.length).toBe(3);
  });

  it("renders copy link and preview buttons for each style", () => {
    render(<StylesCatalog {...defaultProps} />);

    // We should have 3 copy link buttons (one for each style)
    const copyLinkButtons = screen.getAllByTestId("copy-link-button");
    expect(copyLinkButtons.length).toBe(3);

    // We should have 3 eye icons for preview (one for each style)
    const eyeIcons = screen.getAllByTestId("eye-icon");
    expect(eyeIcons.length).toBe(3);
  });
});
