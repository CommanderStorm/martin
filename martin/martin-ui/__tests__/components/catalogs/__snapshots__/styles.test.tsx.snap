// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`StylesCatalog Component matches snapshot for loaded state with mock data 1`] = `
<DocumentFragment>
  <div
    class="space-y-6"
  >
    <div
      class="flex items-center justify-between"
    >
      <div>
        <h2
          class="text-2xl font-bold text-foreground"
        >
          Styles Catalog
        </h2>
        <p
          class="text-muted-foreground"
        >
          Browse and preview all available map styles and themes
        </p>
      </div>
      <div
        class="relative"
      >
        <div
          data-testid="search-icon"
        >
          Search
        </div>
        <input
          class="pl-10 w-64 bg-card"
          placeholder="Search styles..."
          value=""
        />
      </div>
    </div>
    <div
      class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6"
    >
      <div
        class="hover:shadow-lg transition-shadow"
      >
        <div
          data-testid="card-header"
        >
          <div
            class="flex items-center justify-between"
          >
            <div
              class="flex items-center space-x-2"
            >
              <div
                data-testid="brush-icon"
              >
                Brush
              </div>
              <div
                class="text-lg"
                data-testid="card-title"
              >
                Basic Style
              </div>
            </div>
            <span
              data-testid="badge"
              variant="secondary"
            >
              vector
            </span>
          </div>
          <div
            data-testid="card-description"
          >
            /styles/basic.json
          </div>
        </div>
        <div
          data-testid="card-content"
        >
          <div
            class="space-y-4"
          >
            <div
              data-testid="maplibre-map"
              style="width: 100%; aspect-ratio: 1.7777777777777777; border-radius: var(--radius); background-image: linear-gradient(to bottom right, var(--tw-gradient-stops)); background-color: rgb(229, 231, 235);"
            />
            <div
              class="space-y-2 text-sm text-muted-foreground"
            >
              <div
                class="flex justify-between"
              >
                <span>
                  Layers:
                </span>
                <span>
                  10
                </span>
              </div>
              <div
                class="flex justify-between"
              >
                <span>
                  Modified:
                </span>
                <span>
                  15/01/2023, 01:00:00
                </span>
              </div>
            </div>
            <div>
              <p
                class="text-sm font-medium mb-2"
              >
                Color Palette:
              </p>
              <div
                class="flex space-x-1"
              >
                <div
                  class="w-6 h-6 rounded border border-gray-200"
                  style="background-color: rgb(255, 87, 51);"
                  title="#FF5733"
                />
                <div
                  class="w-6 h-6 rounded border border-gray-200"
                  style="background-color: rgb(51, 255, 87);"
                  title="#33FF57"
                />
                <div
                  class="w-6 h-6 rounded border border-gray-200"
                  style="background-color: rgb(51, 87, 255);"
                  title="#3357FF"
                />
                <div
                  class="w-6 h-6 rounded border border-gray-200"
                  style="background-color: rgb(243, 255, 51);"
                  title="#F3FF33"
                />
              </div>
            </div>
            <div
              class="flex space-x-2"
            >
              <button
                class="flex-1 bg-transparent"
                data-testid="copy-link-button"
                link="/style/Basic Style"
                variant="outline"
              >
                Copy Link
              </button>
              <div>
                <div
                  data-testid="tooltip-trigger"
                >
                  <button
                    class="flex-1"
                    disabled=""
                  >
                    <div
                      data-testid="eye-icon"
                    >
                      Eye
                    </div>
                    Preview
                  </button>
                </div>
                <div
                  data-testid="tooltip-content"
                >
                  <p>
                    Not currently implemented in the frontend
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="hover:shadow-lg transition-shadow"
      >
        <div
          data-testid="card-header"
        >
          <div
            class="flex items-center justify-between"
          >
            <div
              class="flex items-center space-x-2"
            >
              <div
                data-testid="brush-icon"
              >
                Brush
              </div>
              <div
                class="text-lg"
                data-testid="card-title"
              >
                Hybrid Style
              </div>
            </div>
            <span
              data-testid="badge"
              variant="secondary"
            >
              hybrid
            </span>
          </div>
          <div
            data-testid="card-description"
          >
            /styles/hybrid.json
          </div>
        </div>
        <div
          data-testid="card-content"
        >
          <div
            class="space-y-4"
          >
            <div
              data-testid="maplibre-map"
              style="width: 100%; aspect-ratio: 1.7777777777777777; border-radius: var(--radius); background-image: linear-gradient(to bottom right, var(--tw-gradient-stops)); background-color: rgb(229, 231, 235);"
            />
            <div
              class="space-y-2 text-sm text-muted-foreground"
            >
              <div
                class="flex justify-between"
              >
                <span>
                  Version:
                </span>
                <span>
                  abc123
                </span>
              </div>
              <div
                class="flex justify-between"
              >
                <span>
                  Layers:
                </span>
                <span>
                  15
                </span>
              </div>
              <div
                class="flex justify-between"
              >
                <span>
                  Modified:
                </span>
                <span>
                  25/03/2023, 01:00:00
                </span>
              </div>
            </div>
            <div
              class="flex space-x-2"
            >
              <button
                class="flex-1 bg-transparent"
                data-testid="copy-link-button"
                link="/style/Hybrid Style"
                variant="outline"
              >
                Copy Link
              </button>
              <div>
                <div
                  data-testid="tooltip-trigger"
                >
                  <button
                    class="flex-1"
                    disabled=""
                  >
                    <div
                      data-testid="eye-icon"
                    >
                      Eye
                    </div>
                    Preview
                  </button>
                </div>
                <div
                  data-testid="tooltip-content"
                >
                  <p>
                    Not currently implemented in the frontend
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="hover:shadow-lg transition-shadow"
      >
        <div
          data-testid="card-header"
        >
          <div
            class="flex items-center justify-between"
          >
            <div
              class="flex items-center space-x-2"
            >
              <div
                data-testid="brush-icon"
              >
                Brush
              </div>
              <div
                class="text-lg"
                data-testid="card-title"
              >
                Satellite Style
              </div>
            </div>
          </div>
          <div
            data-testid="card-description"
          >
            /styles/satellite.json
          </div>
        </div>
        <div
          data-testid="card-content"
        >
          <div
            class="space-y-4"
          >
            <div
              data-testid="maplibre-map"
              style="width: 100%; aspect-ratio: 1.7777777777777777; border-radius: var(--radius); background-image: linear-gradient(to bottom right, var(--tw-gradient-stops)); background-color: rgb(229, 231, 235);"
            />
            <div
              class="space-y-2 text-sm text-muted-foreground"
            />
            <div
              class="flex space-x-2"
            >
              <button
                class="flex-1 bg-transparent"
                data-testid="copy-link-button"
                link="/style/Satellite Style"
                variant="outline"
              >
                Copy Link
              </button>
              <div>
                <div
                  data-testid="tooltip-trigger"
                >
                  <button
                    class="flex-1"
                    disabled=""
                  >
                    <div
                      data-testid="eye-icon"
                    >
                      Eye
                    </div>
                    Preview
                  </button>
                </div>
                <div
                  data-testid="tooltip-content"
                >
                  <p>
                    Not currently implemented in the frontend
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`StylesCatalog Component matches snapshot for loading state 1`] = `
<DocumentFragment>
  <div
    data-testid="catalog-skeleton"
  >
    <div
      data-testid="skeleton-title"
    >
      Styles Catalog
    </div>
    <div
      data-testid="skeleton-description"
    >
      Preview all available map styles and themes
    </div>
  </div>
</DocumentFragment>
`;
